frontends/p4/redundantParsers.h

```diff
@@ -38,27 +39,26 @@ class FindRedundantParsers : public Inspector {
 /** Find .apply() calls on parsers that are on redundantParsers, and
  *  eliminate them.
  */
-class EliminateSubparserCalls : public Transform {
+class EliminateSubparserCalls : public Transform, public ResolutionContext {
     const std::set<const IR::P4Parser *> &redundantParsers;
-    ReferenceMap *refMap;
     TypeMap *typeMap;
     const IR::Node *postorder(IR::MethodCallStatement *methodCallStmt) override;
 
  public:
     EliminateSubparserCalls(const std::set<const IR::P4Parser *> &redundantParsers,
-                            ReferenceMap *refMap, TypeMap *typeMap)
-        : redundantParsers(redundantParsers), refMap(refMap), typeMap(typeMap) {}
+                            TypeMap *typeMap)
+        : redundantParsers(redundantParsers), typeMap(typeMap) {}
 };
 
 class RemoveRedundantParsers : public PassManager {
     std::set<const IR::P4Parser *> redundantParsers;
 
  public:
-    RemoveRedundantParsers(ReferenceMap *refMap, TypeMap *typeMap, const RemoveUnusedPolicy &policy)
-        : PassManager{new TypeChecking(refMap, typeMap, true),
+    RemoveRedundantParsers(TypeMap *typeMap, const RemoveUnusedPolicy &policy)
+        : PassManager{new TypeChecking(nullptr, typeMap, true),
```

Comment:
I missed this in the other PRs but passing in nullptr doesn't seem right to me. Any way to make the constructor safe and just use two parameters? I also ran into this with ConstantFolding, which now has an ambiguous constructor.